libs/pika/threading_base/include/pika/threading_base/thread_num_tss.hpp

```diff
@@ -39,8 +27,57 @@ namespace pika { namespace threads { namespace detail {
         std::size_t thread_pool_num;
     };
 
-    PIKA_EXPORT void set_thread_nums_tss(const thread_nums&);
-    PIKA_EXPORT thread_nums get_thread_nums_tss();
+    inline static thread_local thread_nums thread_nums_tss_ = {
+        std::size_t(-1), std::size_t(-1), std::size_t(-1)};
```

Comment:
Much better, but this needs to stay in the source file. The thread numbers will otherwise be TU-specific.

Reply:
I should probably make the non `detail` namespace functions non inline, since they might be accessed from across a shared library boundary and the standard isn't clear on how an inline var behaves in that case. I'll push another patch just to be safe.

Reply:
TIL! Thanks. However, it needs to be `inline`, not `inline static`.

Reply:
And I'm pretty sure the functions have to stay as `inline`.